src/Goteo/Console/Command/OpenDataCommand.php

```diff
@@ -10,71 +10,320 @@
+                new InputOption('call', 'c', InputOption::VALUE_OPTIONAL, "If specified, extracts data for the given call "),
+                new InputOption('sdg', 's', InputOption::VALUE_IS_ARRAY | InputOption::VALUE_OPTIONAL, 'If specified, extracts data for the given sdgs'),
+                new InputOption('footprint', 'f', InputOption::VALUE_IS_ARRAY | InputOption::VALUE_OPTIONAL, 'If specified, extracts data for the given footprints')
+            ])
+            ->setHelp(<<<EOT
 This command generates files using the data from different sources and saves them. The sources can be channels, matchers, calls or projects.
 
 Usage:
 
 Extract Open Data for a call
 <info>./console opendata --call goteo </info>
 
-Update the provided call's summary data
-<info>./console opendata --call call_id </info>
+Extract Open Data for a Sdg
+<info>./console opendata --sdg 1 </info>
+
+Extract Open Data for multiple Sdgs
+<info>./console opendata -s 1 -s 2 -s 3 </info>
+
+Extract Open Data for a Footprint
+<info>./console opendata --footprint 1 </info>
+
+Extract Open Data for multiple Footprints
+<info>./console opendata -f 1 -f 2 -f 3 </info>
+
 
 EOT
 );
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $call_id  = $input->getOption('call');
 
         $this->log('Extract OpenData info', [], 'info');
 
-        if (isset($call_id)) {
+        if ($callOption = $input->getOption('call')) {
+            $call_id  = $callOption;
             $this->log("Retrieving {$call_id}'s data", [], 'info');
 
-            $call = Call::get($call_id);
+            try {
+                $call = Call::get($call_id);
+            } catch (ModelNotFoundException $e) {
+                $this->log("Call $call_id does not exist", [], 'error');
+                return;
+            }
+
             $this->extractCallOpenData($call);
         }
+
+        if ($listSdg = $input->getOption(('sdg'))) {
+            foreach ($listSdg as $sdg_id) {
+                $this->log("Retrieving {$sdg_id}'s data", [], 'info');
+
+                $sdg = Sdg::get($sdg_id);
+                $this->extractSdgOpenData($sdg);
+            }
+        }
+
+        if ($listFootprints = $input->getOption('footprint')) {
+            foreach ($listFootprints as $footprint_id) {
+                $this->log("Retrieving {$footprint_id}'s data", [], 'info');
+
+                $footprint = Footprint::get($footprint_id);
+                $this->extractFootprintOpenData($footprint);
+            }
+        }
+
+    }
+
+    private function extractSdgOpenData(Sdg $sdg): void {
+
+        $this->extractSdgProjects($sdg);
+        $this->extractSdgInvests($sdg);
+    }
+
+    private function extractSdgProjects(Sdg $sdg) {
+        $fileName = time() . '-' . $sdg->id . '-projects.csv';
+        $file = File::factory(['bucket' => AWS_S3_BUCKET_DOCUMENT]);
+        $file->connect();
+        $file->setPath("open_data/sdg/$sdg->id/projects");
```

Comment:
These 4 lines are repeated in a pattern that we could have extracted as a method that returns a file. Example:

```
function getFile(int $id, string $type)
{
    $fileName = time() . "-$id-$type.csv";
    $file = File::factory(['bucket' => AWS_S3_BUCKET_DOCUMENT]);
    $file->connect();
    $file->setPath("open_data/sdg/$id/$type");
}
```